src/engine/condition.ts

```diff
@@ -56,3 +62,63 @@ const createCondition = (condition: EngineCondition) => (
 };
 
 export default createCondition;
+
+/* Matching conditions pieces */
+
+const isArrayIntersects =
```

Comment:
I think this is getting quite confusing, we have `filters` code:
- `src/engine/filters`
- `src/engine/condition`
- `src/utils`

Can we consolidate and name so it is easy to reason about how they interact?

Makes sense to have `src/engine/filters` export just a single filter for use elsewhere. 

WDYT?

Reply:
I agree that it is getting confusing. The naming is bad too.

I'll study the idea of how to simplify the filter interface.